modules/geo-layers/src/tile-layer/tile-layer.js

```diff
@@ -59,7 +59,7 @@ export default class TileLayer extends CompositeLayer {
         (changeFlags.updateTriggersChanged.all || changeFlags.updateTriggersChanged.getTileData));
 
     if (!tileset) {
-      tileset = new Tileset2D(this._getTilesetOptions(props));
+      tileset = new (this._getTileset2DClass())(this._getTilesetOptions(props));
```

Comment:
This class property business is the part I don't really like. 

Just as a thought experiment, what do we lose if we ask the user to pass in an instance of `Tileset2D`.

Reply:
Passing `Tileset2D` as a prop is the approach #5504 took.

Reply:
I don't have super strong feelings on either approach. One thought: when using the deck.gl/json module having the prop be a class means we can remain more declarative as the class can simply be passed in as a constant: https://deck.gl/docs/api-reference/json/json-converter#constants-catalogs